### PR TITLE
coverage stop areas implementation  

### PR DESCRIPTION
**Pull request by nberard**
runtime local: 768 for 2017-02-17
runtime stat_compiler prod: 30547 for 2017-02-17
no data differences

### CODE REVIEW

**Comment by nberard** [comment]
tests incomming

File: analyzers/coverage_stop_area.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
+import json
+from analyzer import Analyzer
+from datetime import datetime
+
+
+class AnalyzeCoverageStopArea(Analyzer):
+    @staticmethod
+    def get_tuples_from_stat_dict(stat_dict):
+        result = []
+        stop_areas = []
```
Comment (azime) [inline analyzers/coverage_stop_area.py]: use genetrator for "stop_areas"

File: analyzers/requests_calls.py [hunk 2]
```diff
@@ -43,7 +43,7 @@ def collect_data_from_df(self, dataframe):
 
     def get_data(self):
         files = self.get_files_to_analyze()
-        df = self.spark_context.read.json(files)
+        df = self.spark_session.read.json(files)
```
Comment (azime) [inline analyzers/requests_calls.py]: 👍

File: analyzers/coverage_stop_area.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        if 'journeys' in stat_dict and len(stat_dict['journeys']) > 0:
```
Comment (rchoquet) [inline analyzers/coverage_stop_area.py]: Seems to be working but i think you could invert some conditions and add some return/break statements to try to flatten this block of code

Comment (nberard) [inline analyzers/coverage_stop_area.py]: :+1:

File: analyzers/coverage_stop_area.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        return result
+
+    def collect_data_from_df(self, rdd):
+        if rdd.count():
+            coverage_stop_areas = rdd.flatMap(
+                self.get_tuples_from_stat_dict
+            ).reduceByKey(
+                lambda (a), (b): (a + b)
+            ).collect()
+            coverage_stop_areas = [tuple(list(key_tuple) + [nb]) for (key_tuple, nb) in coverage_stop_areas]
```
Comment (azime) [inline analyzers/coverage_stop_area.py]: minor:
```python
return [tuple(list(key_tuple) + [nb]) for (key_tuple, nb) in coverage_stop_areas
```

File: analyzers/coverage_stop_area.py [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+                                        "city_insee": admin_insee,
+                                        "department_code": deparment_code
+                                    }
+
+                                    if stop_area_dict not in stop_areas_of_journey:
+                                        stop_areas.append(stop_area_dict)
+                                        stop_areas_of_journey.append(stop_area_dict)
+                except KeyError, e:
+                    print('key "%s" is missing in dictionnary from json' % e.message)
+                    # print(json.dumps(stat_dict))
```
Comment (azime) [inline analyzers/coverage_stop_area.py]: delete comment

**Pull request merged**